phac-nml/irida-next
# Metadata Listing UI

## What does this PR do and why?
This PR adds the metadata listing on the sample#show view. The view now has tabs for attachments and metadata listings.

Currently, metadata filtering, sorting, and pagination will not be included.

## Screenshots or screen recordings
![image](https://github.com/phac-nml/irida-next/assets/82407232/4aa186eb-1e01-4502-8951-cce66424640c)
![image](https://github.com/phac-nml/irida-next/assets/82407232/d47277f0-c59e-4a09-a645-2950ad425725)


## How to set up and validate locally
1. Add metadata to a sample
2. Navigate to the sample page
3. As `turbo_stream` was modified for much of the page actions, test all page actions and ensure proper redirects and re-renders.
- Upload files
- Concatenate files
- Delete files
- Edit sample
- Remove sample
4. Ensure clicking between the files and metadata tabs list the correct info in the table below
5. Ensure the sample metadata is accurately displayed.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
